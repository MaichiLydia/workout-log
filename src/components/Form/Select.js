import React from "react";
import PropTypes from "prop-types";

function Select(props) {
  function validate(e) {
    const name = e.target.name
    const value = e.target.value

    if (props.required && value.trim() === "") {
      props.onChange(name, value, "Required Field")
      return;
    }

    props.onChange(name, value)
  }
  return (
    <select className="form__select" name={props.name} onChange={validate} id={props.id}>
      {props.options.map(option => (
        <option className="form__option" value={option.id} key={option.id}>
          {option.name}
        </option>
      ))}
    </select>
  );
}

Select.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  options: PropTypes.array.isRequired,
  onChange: PropTypes.func
};

export default Select;