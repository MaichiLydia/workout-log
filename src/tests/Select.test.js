import React from "react";
import Select from "../components/Form/Select"
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';


it('check if select receives the right values', () => {
  const props = {
      id: "cars",
      name: "name",
      required: true,
      options: [{ id: 1, name: "Gol"}, { id: 2, name: "FordFiesta"}, {id:3, name:"Uno"} ],
      onChange: () => true
      }
  const select = mount(<Select {...props}/>)
  expect(toJson(select)).toMatchSnapshot()
});
