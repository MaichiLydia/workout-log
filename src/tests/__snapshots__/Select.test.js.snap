// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check if select receives the right values 1`] = `
<Select
  id="cars"
  name="name"
  onChange={[Function]}
  options={
    Array [
      Object {
        "id": 1,
        "name": "Gol",
      },
      Object {
        "id": 2,
        "name": "FordFiesta",
      },
      Object {
        "id": 3,
        "name": "Uno",
      },
    ]
  }
  required={true}
>
  <select
    className="form__select"
    id="cars"
    name="name"
    onChange={[Function]}
  >
    <option
      className="form__option"
      key="1"
      value={1}
    >
      Gol
    </option>
    <option
      className="form__option"
      key="2"
      value={2}
    >
      FordFiesta
    </option>
    <option
      className="form__option"
      key="3"
      value={3}
    >
      Uno
    </option>
  </select>
</Select>
`;
